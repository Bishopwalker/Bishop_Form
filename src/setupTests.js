// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import {render as rtlRender} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import reducer from './ducks/reducer'
import {BrowserRouter} from "react-router-dom";
import {createMemoryHistory} from "history";
//create a customRender that wraps the UI in a memory Router
// const customRender=(ui,options)=>{
//     return render(ui, {wrapper: MemoryRouter, ...options})
// }
// //re-export everything
// export * from '@testing-library/react'
//
// //override render method
//
// export {customRender as render};

function render(
    ui,
    {   route = '/',
        history = createMemoryHistory({initialEntries: [route]}),
        preloadedState,
        store = configureStore({ reducer: { ...reducer, ...preloadedState } }),
        ...renderOptions
    } ={}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
// re-export everything
export * from '@testing-library/react';

// override render method
export {render};
