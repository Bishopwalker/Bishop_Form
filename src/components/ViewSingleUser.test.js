import React, {useRef} from 'react';
 import {render, cleanup, screen} from'../setupTests'
//import {render,screen} from '@testing-library/react';
import ViewSingleUser from './ViewSingleUser';
import userEvent from "@testing-library/user-event";
import {createMemoryHistory} from 'history'
import {BrowserRouter, MemoryRouter, useLocation} from 'react-router-dom'
import '@testing-library/jest-dom'
import {mocked} from 'jest-mock';

// const renderWithRouter=(ui, {route='localhost:3000/viewSingleUser'} = {}) =>{
//     const history = createMemoryHistory()
//      history.push(route,{state: {
//             firstName: 'Jeff',
//             lastName: 'Dodds',
//             age: 40,
//             hobbies: 'Surfing'
//         }})
//     return render(<Provider store={store}>{ui}</Provider>, {wrapper: BrowserRouter})
// }
// const history = createMemoryHistory()
// const renderWithRouter2=Component=> render(
//     <Router history={history}>
//         <Route component={Component}/>
//     </Router>
// )
jest.mock("react-router-dom",()=>({
    ...jest.requireActual("react-router-dom"),
    useLocation:() => ({
        pathname: 'localhost:3000/viewSingleUser',
    })

}))
jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useRef: jest.fn(),
}));
const useMockedRef = mocked(useRef);
let person = {
    state: {
        firstName: 'Jeff',
        lastName: 'Dodds',
        age: 40,
        hobbies: 'Surfing'
    }
}

describe('View Single User Component', () => {
    function mockAppState() {

    }


    //afterEach(cleanup)
    afterEach(() => {
        jest.clearAllMocks()
        cleanup()
    })
    afterAll(() =>{
        jest.resetAllMocks()
    })


    test('Should take a snapshot', () => {
       // const ref = { current: { people: {people:[{id:3,firstName: 'Jeff', lastName: 'Dodds', age: 40, hobbies: 'Surfing'}]}}}
        //jest.spyOn(React, 'useRef').mockReturnValueOnce({current: {people: {people:[{id:3,firstName: 'Jeff', lastName: 'Dodds', age: 40, hobbies: 'Surfing'}]}}});
        const {asFragment} = render(<ViewSingleUser />)
        expect(asFragment()).toMatchSnapshot()
    })
    test('Should render the correct text', () => {
         render(<ViewSingleUser />)
        expect(screen.getByText(/The List is empty/i)).toBeInTheDocument()
       // expect(screen.getByText(/Age/i)).toBeInTheDocument()
       // expect(screen.getByText(/Hobbies/i)).toBeInTheDocument()
    })
})