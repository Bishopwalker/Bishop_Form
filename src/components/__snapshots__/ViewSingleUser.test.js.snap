// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Single User Component Should take a snapshot 1`] = `
<DocumentFragment>
  <div>
     
    <h3>
      The List is Empty
    </h3>
  </div>
</DocumentFragment>
`;
